cpp/src/arrow/compute/exec/test_util.h

```diff
@@ -145,5 +145,38 @@ class Random64Bit {
   std::uniform_int_distribution<uint64_t> dist_;
 };
 
+/*
+  Specify properties of a table to be generated.
+    - num_ids is the number of unique keys in the table
+    - time_frequency indicates the amount of time between data points that lie between
+  start and end (inclusive)
+    - num_columns indicates the amount of random columns in the table
+    - column_prefix specifies the prefix each randomly generated column should have
+    - seed is the random seed the random array generator is given to generate the random
```

Comment:
These can be comments on the fields themselves.  See `AsofJoinNodeOptions` in `arrow/compute/exec/options.h`